File: src/main/java/com/example/demo/Service/VideoService.java
package com.example.demo.Service;

import com.example.demo.DAO.DAO;
import com.example.demo.Model.Video;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Service class that insures the link between the application methods and the database.
 */
@Service
public class VideoService {
    private DAO<Video> dao;

    @Autowired
    public VideoService(DAO<Video> dao)
    {
        this.dao = dao;
    }

    /**
     * Sends the request to get all the Videos from the database.
     * @return List[Videos]
     */
    public List<Video> getVideos() {
        return dao.getAll();
    }

    /**
     * Sends the request to get the Video with the specified id.
     * @param id search criteria
     * @return Client
     */
    public Video getVideoByID(Long id) {
        return dao.findByID(id);
    }

    /**
     * Sends the request to get the Video with the specified name.
     * @param name search criteria
     * @return Video
     */
    public List<Video> getVideoByName(String name) {
        return dao.findByName(name);
    }

    /**
     * Sends the request to get all the Videos posted by a specified channel.
     * @param channel search criteria
     * @return List[Video]
     */
    public List<Video> getVideosByChannel(String channel) {
        return dao.findAllByChannel(channel);
    }

    /**
     * Sends the request to get all the Videos of a specified genre.
     * @param genre search criteria
     * @return List[Video]
     */
    public List<Video> getVideosByGenre(String genre) {
        return dao.findAllByGenre(genre);
    }

    /**
     * Sends the request to get all the Videos that start at a specified hour.
     * @param startHour search criteria
     * @return List[Video]
     */
    public List<Video> getVideosByStartHour(int startHour) {
        return dao.findAllByStartHour(startHour);
    }

    /**
     * Sends the Video to be added to the database.
     * @param video the entry to be added
     */
    public void saveVideo(Video video) {
        dao.save(video);
    }

    /**
     * Sends the new Video object to update the database.
     * @param video the entry containing the new information
     */
    public void updateVideo(Video video) {
        dao.update(video);
    }

    /**
     * Sends the id of the Video to be deleted from the database.
     * @param id delete criteria
     */
    public void deleteVideoByID(Long id) {
        dao.delete(id);
    }

}

File: src/app/services/video.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { VideoModel } from '../models/video.model';
import { VideoUploadModel } from '../models/video-upload.model';

@Injectable({
	providedIn: 'root'
})
export class VideoService {

	constructor(private httpClient: HttpClient) { }

	public getAllVideos() {
		return this.httpClient.get<VideoModel>(`/api/getVideos`);
	}

	public getVideoByID(id: number) {
		return this.httpClient.get<VideoModel>(`/api/getVideoByID/${id}`);
	}

	public getVideoByName(name: string) {
		return this.httpClient.get<VideoModel>(`/api/getVideoByName/${name}`);
	}

	public getVideoByChannel(channel: string) {
		return this.httpClient.get<VideoModel>(`/api/getVideosByChannel/${channel}`);
	}

	public getVideoByGenre(genre: string) {
		return this.httpClient.get<VideoModel>(`/api/getVideosByGenre/${genre}`);
	}

	public getVideoByHour(startHour: number) {
		return this.httpClient.get<VideoModel>(`/api/getVideosByHour/${startHour}`);
	}

	public upload(videoData: VideoUploadModel) {
		return this.httpClient.post<VideoUploadModel>(`/api/addVideo`, videoData);
	}

	public update(videoData: VideoModel) {
		return this.httpClient.put<VideoModel>(`/api/updateVideo`, videoData);
	}

	public delete(id: number) {
		return this.httpClient.delete<VideoModel>(`/api/deleteVideoByID/${id}`);
	}

}
